Dockerfile

```diff
@@ -22,4 +21,33 @@ ENV PYTHONPATH=/app
 COPY agent /app/agent
 COPY ostorlab.yaml /app/agent/ostorlab.yaml
 WORKDIR /app
+
+FROM base AS builder
+
+WORKDIR /tmp
+
+ARG NMAP_VERSION=7.95
+
+RUN apt-get update && apt-get install -y \
+    wget \
+    automake \
+    python3-venv \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN python3 -m pip install build
+
+RUN wget https://nmap.org/dist/nmap-${NMAP_VERSION}.tar.bz2 \
+    && tar jxvf nmap-${NMAP_VERSION}.tar.bz2 \
+    && cd /tmp/nmap-${NMAP_VERSION} \
+    && ./configure \
+    && make \
+    && sudo make install \
+    && cd / \
+    && rm -rf /tmp/nmap-$(NMAP_VERSION)
+
+FROM base
```

Comment:
Usually the steps that change less are executed first, so you can benefit from caching.. and steps that change the most like the agent code .. should be at the end. check [Image-building best practices](https://docs.docker.com/get-started/workshop/09_image_best/) for more details.
Also, why not install nmap in the same initial stage?

Reply:
@deadly-panda by initial stage, you mean base stage? without using builder stage?
if that's the case, it's to avoid shipping with the build dependencies

Reply:
I took the `COPY` commands to the end, since these are the ones that are more likely ruin caching